webapp/src/snippetBuilder.tsx

```diff
@@ -101,59 +134,86 @@ export class CreateSnippetBuilder extends data.Component<ISettingsProps, CreateS
         });
     }
 
+    /**
+     * Closes the modal
+     */
     cancel() {
         pxt.tickEvent("snippetBuilder.cancel", undefined, { interactiveConsent: true });
         this.hide();
     }
 
+    /**
+     * Takes the output from state, runs replace tokens, decompiles the resulting typescript
+     * and outputs the result as a Blockly xmlDOM. This then uses appendDomToWorkspace to attach 
+     * our xmlDOM to the mainWorkspaces passed to the component.
+     */
     injectBlocksToWorkspace() {
-        const { mainWorkspace, output } = this.state;
+        const { mainWorkspace, tsOutput } = this.state;
 
         compiler.getBlocksAsync()
-            .then(blocksInfo => compiler.decompileBlocksSnippetAsync(this.replaceTokens(output), blocksInfo))
+            .then(blocksInfo => compiler.decompileBlocksSnippetAsync(this.replaceTokens(tsOutput), blocksInfo))
             .then(resp => {
-                Blockly.Xml.appendDomToWorkspace(Blockly.Xml.textToDom(resp), mainWorkspace);
+                const xmlDOM = Blockly.Xml.textToDom(resp)
+                Blockly.Xml.appendDomToWorkspace(xmlDOM, mainWorkspace);
             });
     }
 
+    /**
+     * Hides modal and injects our blocks to the work space.
+     */
     confirm() {
         this.injectBlocksToWorkspace();
         this.hide();
     }
 
+    /** 
+     * @param increment - this adds either 1 or -1 to the value of currentQuestion
+     */
     changePage(increment: 1 | -1) {
         const { currentQuestion } = this.state;
         this.setState({ currentQuestion: currentQuestion + increment });
     }
 
+    /**
+     * Changes page by 1 if next question exists.
+     * Looks for output and appends the next questions output if it exists and
+     * is not already attached to the current output.
+     */
     nextPage() {
         const { config } = this.state;
-        const { currentQuestion, output } = this.state;
+        const { currentQuestion, tsOutput } = this.state;
         const nextQuestion = config.questions[currentQuestion + 1];
-        // If output exists
-        if (nextQuestion.output) {
+        // If next question exists
+        if (nextQuestion) {
             // If output is not already appended
-            if (output.indexOf(nextQuestion.output) === -1) {
-                this.setState({ output: `${output}\n${nextQuestion.output}`});
+            if (nextQuestion.output && tsOutput.indexOf(nextQuestion.output) === -1) {
+                this.setState({ tsOutput: `${tsOutput}\n${nextQuestion.output}`});
             }
             // Change page to page + 1
             this.changePage(1);
         }
     }
 
+    /**
+     * Calls changePage with a -1 decrementor
```

Comment:
It's great that you added comments but some of them might be too much :) This one for example doesn't describe anything that's not obvious in the code and it's also too specific to the code. Generally the comments shouldn't prescribe an implementation but rather tell the goals and the usage patterns.

Reply:
not every function needs comments

Reply:
If you need a good reference on stuff like this, Clean Code is very popular and good:
https://www.amazon.com/Clean-Code-Handbook-Software-Craftsmanship/dp/0132350882